### PR TITLE
mathematical definition of backward method

### PR DESCRIPTION
**Pull request by scheidan**
It took me quite a while to figure out what _exactly_ the `backward` methods suppose to compute, so I added a short definition. Maybe that is obvious, but I have more a math than a CS background.

PS: I don't know how I can test the latex formating

### CODE REVIEW

**Comment by EdwardRaff** [comment]
I have a CS background and it took me a while too, so I think it's a good idea :)

File: docs/source/tutorial/function.rst [hunk 1]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
 This example just implements the rule.
 Look at the return values, the function just packs the gradient of each input in same order and returns them.
 
-By just defining the core computation of forward and backward, Function class provides a chaining logic on it (i.e. storing the history of computation, etc.).
+By just defining the core computation of forward and backward,
+Function class provides a chaining logic on it (i.e. storing the
+history of computation, etc.).
+
+.. note::
+   Assuming we implement a (forward) function :math:`f(x)=y` which takes as input the
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: I think `y = f(x)` is more common.

File: docs/source/tutorial/function.rst [hunk 2]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
...
+   vector :math:`x \in \mathbb{R}^n` and produces as output a vector
+   :math:`y \in \mathbb{R}^m`. Then the ``backward`` method has to
+	 compute
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Remove tab and align with previous line.

File: docs/source/tutorial/function.rst [hunk 3]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
...
+   .. math::
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Add empty line between this line and the previous line.

File: docs/source/tutorial/function.rst [hunk 4]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
...
+      \lambda_i = \sum_{j=1}^m \frac{\partial y_j}{\partial x_i} \,
+      \gamma_j \,\, \text{for} i = 1 \dots n
+    where :math:`\gamma` are the ``grad_outputs``. Note, that the
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Add an empty line between this line and the previous line.

File: docs/source/tutorial/function.rst [hunk 5]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
...
+history of computation, etc.).
+
+.. note::
+   Assuming we implement a (forward) function :math:`f(x)=y` which takes as input the
+   vector :math:`x \in \mathbb{R}^n` and produces as output a vector
+   :math:`y \in \mathbb{R}^m`. Then the ``backward`` method has to
+	 compute
+   .. math::
+      \lambda_i = \sum_{j=1}^m \frac{\partial y_j}{\partial x_i} \,
+      \gamma_j \,\, \text{for} i = 1 \dots n
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Add `\,` after `for`

File: docs/source/tutorial/function.rst (hunk 4 above)
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Remove one space in the head of this line to align the line with the previous paragraph.

File: docs/source/tutorial/function.rst [hunk 6]
```diff
@@ -86,7 +86,22 @@ Recall the rule of differentiation of multiplication.
...
+    where :math:`\gamma` are the ``grad_outputs``. Note, that the
+    resulting vector :math:`lambda` must have the same shape as the inputs of the ``forward`` method.
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: ditto (Remove one space).

File: docs/source/tutorial/function.rst (hunk 4 above)
Comment (delta2323) [inline docs/source/tutorial/function.rst]: I think ":math:`\gamma_j`'s are ..." is more appropriate.

File: docs/source/tutorial/function.rst (hunk 6 above)
Comment (delta2323) [inline docs/source/tutorial/function.rst]: Add a backslash in front of "lambda".

**Comment by delta2323** [comment]
I added comments. Please check them.

**Comment by scheidan** [comment]
I tried to address you comments. I'm not very familiar with the GitHub workflow, hope I didn't mess up anything :)

File: docs/source/tutorial/function.rst [hunk 7]
```diff
@@ -86,7 +86,23 @@ Recall the rule of differentiation of multiplication.
...
+.. note::
+   Assuming we implement a (forward) function :math:`y=f(x)` which takes as input the
+   vector :math:`x \in \mathbb{R}^n` and produces as output a vector
+   :math:`y \in \mathbb{R}^m`. Then the ``backward`` method has to compute
+
+   .. math::
+      \lambda_i = \sum_{j=1}^m \frac{\partial y_j}{\partial x_i} \,
+      \gamma_j \,\, \text{for}\, i = 1 \dots n
+
+   where :math:`\gamma_j` are the ``grad_outputs``. Note, that the
```
Comment (delta2323) [inline docs/source/tutorial/function.rst]: where :math:`\gamma` **is**

**Comment by delta2323** [comment]
LGTM except one comment (sorry, this is where I suggested you to change before).

**Comment by scheidan** [comment]
Fixed

**Comment by delta2323** [comment]
LGTM :+1:

**Comment by delta2323** [comment]
Thank you for your PR!

**Pull request merged**